pygpu/tests/test_basic.py

```diff
@@ -7,31 +7,33 @@
 
 
 def test_tril():
-    for shape in [(10, 5), (5, 10), (10, 10)]:
-        for order in ['c', 'f']:
-            for inplace in [True, False]:
-                ac, ag = gen_gpuarray(shape, 'float32',
-                                      order=order, ctx=context)
-                result = tril(ag, inplace=inplace)
-                assert numpy.all(numpy.tril(ac) == result)
-                if inplace:
-                    assert numpy.all(numpy.tril(ac) == ag)
-                else:
-                    assert numpy.all(ac == ag)
+    for dtype in ['float32','float64']:
+        for shape in [(10, 5), (5, 10), (10, 10)]:
+            for order in ['c', 'f']:
+                for inplace in [True, False]:
+                    ac, ag = gen_gpuarray(shape, dtype,
+                                          order=order, ctx=context)
+                    result = tril(ag, inplace=inplace)
+                    assert numpy.all(numpy.tril(ac) == result)
+                    if inplace:
+                        assert numpy.all(numpy.tril(ac) == ag)
+                    else:
+                        assert numpy.all(ac == ag)
 
 
 def test_triu():
-    for shape in [(10, 5), (5, 10), (10, 10)]:
-        for order in ['c', 'f']:
-            for inplace in [True, False]:
-                ac, ag = gen_gpuarray(shape, 'float32',
-                                      order=order, ctx=context)
-                result = triu(ag, inplace=inplace)
-                assert numpy.all(numpy.triu(ac) == result)
-                if inplace:
-                    assert numpy.all(numpy.triu(ac) == ag)
-                else:
-                    assert numpy.all(ac == ag)
+    for dtype in ['float32','float64']:
+        for shape in [(10, 5), (5, 10), (10, 10)]:
+            for order in ['c', 'f']:
+                for inplace in [True, False]:
+                    ac, ag = gen_gpuarray(shape, dtype,
+                                          order=order, ctx=context)
+                    result = triu(ag, inplace=inplace)
+                    assert numpy.all(numpy.triu(ac) == result)
```

Comment:
Maybe assert that results have the right dtype? The current test could pass if the value have the wrong dtype I think. Otherwise, it loooks good. Thanks.